test/javascript-wrapper-mocha.js

```diff
@@ -0,0 +1,203 @@
+                handler = callback;
+                commonTest.sendData(component, 'x', "test input");
+            });
+        }).should.become( 'success' );
+    });
+
+    it("should trigger updater function with no input parameters", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                in: {required: true},
+            },
+            updater: function() {
+               handler('success');
+            }
+        }).then(jswrapper).then(commonTest.createComponent).then(function(component){
+            return new Promise(function(callback){
+                handler = callback;
+                commonTest.sendData(component, 'in', "test input");
+            });
+        }).should.become( 'success' );
+    });
+
+    it("should trigger updater function with state as input parameter when cannot introspect", function() {
+
+        var handler;
+        // bind function to the current context and pass argument "testArg"; this function cannot be introspected 
+        var updater = (function(context,arg1){ 
+             arguments.should.have.length(2);
+             arguments[0].should.deep.equal('testArg');
+             arguments[1].should.deep.equal( { in:'test input' } );
+             handler('success');
+        }).bind(this,"testArg");
+
+        return Promise.resolve({
+            inPorts:{
+                in: {required: true}
+            },
+            updater: updater 
+        }).then(jswrapper).then(commonTest.createComponent).then(function(component){
+            return new Promise(function(callback){
+                handler = callback;
+                commonTest.sendData(component, 'in', "test input");
+            });
+        }).should.become( 'success' );
+    });
+
+    it("should trigger updater function with input parameters", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                in1: { datatype: 'string', 
+                      description: "input 1 description",
+                      required: true },
+                in2: { datatype: 'string', 
+                      description: "input 2 description",
+                      required: true }
+            },
+            updater: function( in1, in2 ) {
+               handler(in1+in2);
+            }
+        }).then(jswrapper).then(commonTest.createComponent).then(function(component){
+            return new Promise(function(callback){
+                handler = callback;
+                commonTest.sendData(component, 'in1', "alpha and ");
+                commonTest.sendData(component, 'in2', "omega");
+            });
+        }).should.become( 'alpha and omega' );
+    });
+
+  // This introspect tests must be run in test mode with NODE_ENV=test 
+  // e.g., NODE_ENV=test mocha since we are testing a private function
+  describe('#introspect - run with NODE_ENV=test to fully execute these tests', function() {
```

Comment:
I would if I thought it would be more widely used.  This is a custom-to-javascript introspector though, so I don't see other wrappers as likely to use it.  Also, it's trivial to simply do 
NODE_ENV=test mocha 
to run the test suite.   That is the default way that many teams have their unit testing configured to execute anyway.